qa/tasks/mon_thrash.py

```diff
@@ -257,6 +264,27 @@ def do_thrash(self):
             # Allow successful completion so gevent doesn't see an exception.
             # The DaemonWatchdog will observe the error and tear down the test.
 
+    def switch_task(self):
+        """
+        Pause mon thrasher till other thrashers are done with their iteration.
+        This would help to sync between multiple thrashers, like:
+        1. thrasher-1 and thrasher-2: thrash daemons in parallel
+        2. thrasher-1 and thrasher-2: revive daemons in parallel 
+        This allows us to run some checks after each thrashing and reviving iteration.
+        """
+        if not hasattr(self, 'is_done'):
+            return
+        self.is_done.set()
+        thrashers = self.ctx.ceph[self.config.get('cluster')].thrashers
+        for t in thrashers:
+            if not isinstance(t, MonitorThrasher) and hasattr(t, 'is_done') and (
```

Comment:
I'd suggest adding a 'switch_thrasher' attribute instead of using 'is_done' as a proxy for 'switching between thrashers being configured'.

Also suggest making mon_thrash's stopping variable an Event as well so you don't need to check its type - most other thrashers use an Event for this already